src/main/java/practice/StreamPractice.java

```diff
@@ -23,7 +35,16 @@ public int findMinEvenNumber(List<String> numbers) {
      * But before that subtract 1 from each element on an odd position (having the odd index).
      */
     public Double getOddNumsAverage(List<Integer> numbers) {
-        return 0D;
+
+        AtomicInteger index = new AtomicInteger();
+
+        return Arrays.stream(numbers.toArray())
+                .mapToInt(x -> (int) x)
+                .map(e -> index.getAndIncrement() % 2 == 0 ? e : e - 1)
+                .filter(e -> e % 2 != 0)
+                .mapToDouble(e -> e)
```

Comment:
You are iterating through your range as an analog of for-each loop but it's replaced with lambda, where 
```
for (Integer i : number)
   if (i % 2 == 1)
       return numbers.get(i);
```
is replaced by this reference `i -> i ...numbers.get(i);`
```suggestion
                .map(e -> e % 2 == 0 ? numbers.get(e) : numbers.get(e) - 1)
                .filter(e -> e % 2 != 0)
```